flutter/lib/providers/superformula_provider.dart

```diff
@@ -0,0 +1,227 @@
+//Flutter packages
+import 'dart:convert';
+//Third party packages
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+//My Packages
+import '../utils.dart';
+import '../values.dart' as values;
+
+enum DOWNLOAD_STATE{
+  DOWNLOADING,
+  DOWNLOADED,
+  ERROR,
+}
+
+class SuperFormulaProvider with ChangeNotifier{
+
+  //Object declarations
+
+  DOWNLOAD_STATE qrCodeDownloadingState = DOWNLOAD_STATE.DOWNLOADING;
+
+  String _seed = "";
+
+  String get seed{
+    return _seed;
+  }
+
+  //Network Functions
+
+  Dio dio = Dio(BaseOptions(
+    baseUrl: values.baseUrl,
+    connectTimeout: 8000,
+    receiveTimeout: 8000,
+  ));
+
+  Future getSeed({required bool firstTime,required BuildContext context}) async{
```

Comment:
**thought**: Needing to pass in a context into your provider or view models is an indication that you may be too tightly coupled with your ui code.